### ISSUE TITLE
Setting allowImplicit: false option on aria-allowed-roles fails all roles

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#3213]
Consider the following code:

```html
<div role="button">Button</div>
```

This passes `aria-allowed-role`. But if you configure the rule to not allow implicit roles, it now fails

```js
axe.configure({
  checks: [
    {
      id: 'aria-allowed-role',
      options: {
        allowImplicit: false
      }
    }
  ]
});
```

The reason is that the [final `if` statement in the check](https://github.com/dequelabs/axe-core/blob/develop/lib/commons/aria/get-element-unallowed-roles.js#L91-L100) will _always_ pass unless the element is a `tr` node inside a `grid`.

**Comment by padmavemulapati** [comment]
Validated with the latest develop branch code base,
verified for `aria-roles` ,
```
<div role="button">Button</div>
  <div id="target" role="doc-fizzbuzz">Contents</div>
  <div id="target" role="foo widget melon button">Contents</div>
  <div id="target" role="button melon widget">Contents</div>
  <main><header role="banner" id="pass-header-banner">ok</header></main>
<main><footer role="contentinfo" id="pass-footer-contentinfo">ok</footer></main>
```
<img width="1265" alt="image" src="https://user-images.githubusercontent.com/47409579/187356075-4bc93e28-61ce-482e-bd23-fd2cbd5a2af8.png">
